tests/functional/tests/basic/test_offset.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright(c) 2024 Huawei Technologies
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+from pyocf.types.cache import Cache
+from pyocf.types.core import Core
+from pyocf.types.data import Data, DataSeek
+from pyocf.types.io import IoDir, Sync
+from pyocf.types.volume import RamVolume
+from pyocf.types.volume_core import CoreVolume
+from pyocf.utils import Size as S
+
+
+def test_data_with_offset(pyocf_ctx):
+    cache_device = RamVolume(S.from_MiB(50))
+    core_device = RamVolume(S.from_MiB(50))
+
+    cache = Cache.start_on_device(cache_device)
+    core = Core.using_device(core_device)
+    queue = cache.get_default_queue()
+
+    cache.add_core(core)
+    core_volume = CoreVolume(core)
+    core_volume.open()
+
+    # write using data with offset
+    B1 = b'12345'
+    B2 = b'67890'
+    data = Data(8192)
+    data.seek(DataSeek.BEGIN, 0)
+    data.write(B1, len(B1))
+    data.seek(DataSeek.BEGIN, 4096)
+    data.write(B2, len(B2))
+
+    addr = 4096
+    length = 4096
+    offset = 4096
+    io = core_volume.new_io(queue, addr, length, IoDir.WRITE, 0, 0)
+    io.set_data(data, offset)
```

Comment:
Shouldn't the addr or offset be 0 if we want to perform a normal read?

Reply:
Not quite. The test prepares an 8k buffer but instead of writing its whole content, it writes only its second half by setting offset to 4k.